crates/zksync/core/src/vm/tracer.rs

```diff
@@ -40,6 +40,10 @@ pub struct CheatcodeTracerContext<'a> {
     pub mocked_calls: HashMap<Address, BTreeMap<MockCallDataContext, MockCallReturnData>>,
     /// Expected calls recorder.
     pub expected_calls: Option<&'a mut ExpectedCallTracker>,
+    /// Recorded reads
+    pub recorded_reads: Arc<Mutex<Option<Access>>>,
```

Comment:
Can directly use `Option<&'a mut RecordAccess>` after we move it to `cheatcodes/common`